src/full/Agda/TypeChecking/Monad/Closure.hs

```diff
@@ -12,9 +12,10 @@ enterClosure :: (MonadTCEnv m, ReadTCState m, LensClosure a c)
              => c -> (a -> m b) -> m b
 enterClosure c k | Closure _sig env scope cps x <- c ^. lensClosure = do
   isDbg <- viewTC eIsDebugPrinting
+  isElab <- viewTC eCurrentlyElaborating
   withScope_ scope
     $ locallyTCState stModuleCheckpoints (const cps)
-    $ withEnv env{ envIsDebugPrinting = isDbg }
+    $ withEnv env{ envIsDebugPrinting = isDbg, envCurrentlyElaborating = isElab }
```

Comment:
Here I feel like I am doing the wrong thing but I don't know of the proper way to solve this problem. The problem being that the flag `envCurrentlyElaborating` is supposed to store whether we are currently in the process of executing an elaborate-and-give command however putting it in the `TCEnv` means it gets overwritten when we enter the closure of a metavariable, which is not what we want. But putting it in the `TCState` instead feels even worse since it is just a "local" thing. Maybe @UlfNorell or @andreasabel knows of a better way?

Reply:
I found a better solution (see commit below)